libs/dal/src/repositories/base-repository.ts

```diff
@@ -105,11 +105,15 @@ export class BaseRepository<T> {
     };
   }
 
-  protected mapEntity(data: any): T {
-    return plainToInstance<T, T>(this.entity, JSON.parse(JSON.stringify(data))) as any;
+  protected mapEntity(data: any): T_Response {
+    return plainToInstance<T_Response, T_Response>(this.entity, JSON.parse(JSON.stringify(data))) as any;
   }
 
-  protected mapEntities(data: any): T[] {
-    return plainToInstance<T, T[]>(this.entity, JSON.parse(JSON.stringify(data)));
+  protected mapEntities(data: any): T_Response[] {
+    return plainToInstance<T_Response, T_Response[]>(this.entity, JSON.parse(JSON.stringify(data)));
   }
 }
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const Omit = <T, K extends keyof T>(Class: new () => T, keys: K[]): new () => Omit<T, typeof keys[number]> =>
```

Comment:
Are you sure this is not provided globally by TS? From what I recall Omit is a utility available globally: 

https://www.typescriptlang.org/docs/handbook/utility-types.html#omittype-keys